Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -292,4 +319,47 @@ async function updateArmMetadata(SPN, webAppName: string, resourceGroupName: str
     await azureRESTUtility.updateAzureRMWebAppMetadata(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName, metadata);
 }
 
-run();
+async function updateStartupCommandAndRuntimeStack(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string, runtimeStack: string, startupCommand: string) {
+    try {
+        var configDetails = await azureRESTUtility.getAzureRMWebAppConfigDetails(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName);
+        var linuxFxVersion: string = configDetails.properties.linuxFxVersion;
+        var appCommandLine: string = configDetails.properties.appCommandLine;
+
+        var updatedConfigDetails: string = "";
+        
+        if (!(!!appCommandLine == !!startupCommand && appCommandLine == startupCommand)
+            || runtimeStack != linuxFxVersion) {
+            updatedConfigDetails = JSON.stringify({
+                "properties": {
+                    "linuxFxVersion": runtimeStack,
+                    "appCommandLine": startupCommand
+                }
+            });
+
+            tl.debug("Updating webConfig with the details: " + updatedConfigDetails);
+
+            await azureRESTUtility.updateAzureRMWebAppConfigDetails(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName, updatedConfigDetails);
+
+            console.log("SuccessfullyUpdatedRuntimeStackAndStartupCommand");
+        }
+    }
+    catch (error) {
+        tl.error(tl.loc("FailedToUpdateRuntimeStackAndStartupCommand", error));
+    }
+}
+
+async function updateAppSetting(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string, appSettings: any) {
+    if (appSettings) {
+        try {
+            tl.debug("Updating app settings for builtin images");
+
+            await updateAppSettings(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName, appSettings);
+            
+            console.log(tl.loc("SuccessfullyUpdatedWebAppSettings"));
+        }
+        catch (error) {
+            tl.error(tl.loc("FailedToUpdateAppSettingsInConfigDetails", error));
```

Comment:
Will tl.error fail the task or should you rather throw. Can you check? Applies to method above also.